packages/apps/ds4ch/pages/index.vue

```diff
@@ -8,33 +8,53 @@ import landingPageQuery from "@/graphql/queries/landingPage.graphql";
 const contentful = inject("$contentful");
 const { localeProperties } = useI18n();
 
-const { data: page } = await useAsyncData("homePage", async () => {
-  const variables = {
-    identifier: "/",
-    locale: localeProperties.value.language,
-  };
-
-  const response = await contentful.query(landingPageQuery, variables);
-  return response.data?.landingPageCollection?.items?.[0] || {};
+const { data: contentfulResponse, error: contentfulErrors } =
+  await useAsyncData("homePage", async () => {
+    const variables = {
+      identifier: "/",
+      locale: localeProperties.value.language,
+    };
+
+    return contentful.query(landingPageQuery, variables);
+  });
+
+const page = computed(() => {
+  return contentfulResponse.value?.data?.landingPageCollection?.items?.[0];
 });
 
-const sections = page.value.hasPartCollection?.items.filter((item) => !!item);
+const error = computed(() => {
+  if (
+    !contentfulErrors.value &&
+    !contentfulResponse.value?.data?.landingPageCollection?.items?.[0]
+  ) {
+    showError({ statusCode: 404, statusMessage: "Not Found" });
+  }
+  return contentfulErrors.value;
+});
 
-annotateParity(deepFindEntriesOfType(sections, "ImageCard"));
+const sections = computed(() => {
+  let availableSections = page?.value?.hasPartCollection?.items.filter(
+    (item) => !!item,
+  );
+  annotateParity(deepFindEntriesOfType(availableSections, "ImageCard"));
```

Comment:
Does this really belong inside a computed fn? It seems like it should be in a watcher.